src/main/scala/strawman/collection/mutable/ArrayDeque.scala

```diff
@@ -0,0 +1,491 @@
+package strawman
+package collection.mutable
+
+import scala._
+import scala.collection.{generic, mutable}
```

Comment:
This PR was originally meant for the Scala repo - I unfortunately do not have the time to fully understand strawman.collection (no design docs, fast changing code etc). I assume it should be fairly trivial to adapt since we don't expect massive public API breakage?